opendatahub-io/distributed-workloads
# Add an e2e test for running instructlab distributed training on RHOAI

## Description

[RHOAIENG-14225](https://issues.redhat.com/browse/RHOAIENG-14225) - Automate an E2E test scenario for running instructlab distributed training on RHOAI
Source : https://github.com/redhat-et/ilab-on-ocp/tree/main/standalone

## How Has This Been Tested?




**Pre-requisites** :  [Document ](https://docs.google.com/document/d/15WdoJb9MYcljn5osojjw0NimHxzbuEvQhmTIBIJohko/edit?usp=sharing)
1. Login to cluster and install RHOAI operator with Training Operator component enabled
2. Install `NFS Provisioner`,` Redhat Authorino` and `Redhat Openshift Serverless` operators from OperatorHub 
3. Create  storage class to setup a storage with ReadWriteMany access : 
     Login to cluster and execute below commands
     ```
$ git clone https://github.com/OCP-on-NERC/nerc-ocp-config.git && cd nerc-ocp-config
$ kubectl apply -k  csi-driver-nfs/overlays/nerc-ocp-test
$ kubectl apply -k  nfs/overlays/nerc-ocp-test
    ```
5. Create test-namespace and Service-account linked with it
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: test-sa
  namespace: test-namespace
```
7. Create model server using single model serving platform, ` VLLM Serving  Runtime for Kserve ` serving runtime with accelerator( only supported on GPU- In my case used `NVIDIA A100-SXM4-40GB` for testing ) and the storage bucket data connection. (Check Model route and token authentication enabled with above created service account)
  ![image](https://github.com/user-attachments/assets/b147cefd-5502-4260-9f02-400b0b09c03d)
 Note: If unable to use accelerator profile while deploying model server, check whether you have GPU available in cluster and once gpu is deployed and driver are installed on node then either create accelerator profile or run `oc delete configmap migration-gpu-status -n redhat-ods-applications;oc rollout restart deployment.apps/rhods-dashboard -n redhat-ods-applications`

8. Setup an S3 bucket consisting an tar archive of (SDG-Data, Model and Taxonomy)

```
>> export ODH_NAMESPACE=redhat-ods-applications \
AWS_ACCESS_KEY_ID=<secret-access-key> \
AWS_SECRET_ACCESS_KEY=<s3-secret-key> \
AWS_STORAGE_BUCKET=myrhelai \
AWS_DEFAULT_REGION=us-east-1 \
AWS_DEFAULT_ENDPOINT=https://s3.us-east-1.amazonaws.com \
SDG_OBJECT_STORE_DATA_KEY=rhelai.tar.gz \
SDG_OBJECT_STORE_VERIFY_TLS=True \
JUDGE_ENDPOINT=https://<model-name>-<namespace-name>.apps.<cluster-base-url>.com \
JUDGE_NAME=judge-eval-model \
TEST_NAMESPACE=rhelai-ilab-dw-training-test \
TEST_SERVICE_ACCOUNT=<created-service-account-name> \
JUDGE_SERVING_MODEL_API_KEY=<model-server-service-account-token-secret>

```
```
>> go test timeout 180m -v ./tests/odh -run TestInstructlabTrainingOnRhoai
```

![image (157)](https://github.com/user-attachments/assets/4d06a05b-de27-4529-9107-b77267072d21)


## Merge criteria:
